person.rb

```diff
@@ -0,0 +1,35 @@
+require './decorator'
+require './rentals'
+
+class Person < Nameable
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    @id = Random.rand(1...1999)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+
+    @rentals = []
+    super()
+  end
+
+  def correct_name
+    @name = name
+  end
+
+  attr_reader :id, :rentals
+  attr_accessor :name, :age
+
+  def of_age?
+    @age >= 18
+  end
+
+  def can_use_services?
+    of_age?
+  end
+
+  def add_rentals=(book, date)
+    Rental.new(book, self, date)
+  end
```

Comment:
- Please note that this method should implement the `has-many` relationship between the `Person` class and `Rental` `class, it does not serve as a setter method. Please fix this 🙏